src/collector.rs

```diff
@@ -51,6 +51,19 @@ impl Collector {
         self.raw.get(command).unwrap().clone()
     }
 
+    // returns a string representation of the first capture group of the regex
+    pub fn get_regex_from_command(&mut self, command: &str, regex: &str) -> Option<String> {
```

Comment:
`parse_from_command`, as it is returning the parsed output, not the regex itself

Reply:
Return a `Result<String, Error>`, so we can use `?` and send the error upwards, or return a value.